### ISSUE TITLE
[random] Search crashed

### ISSUE THREAD

**Issue by del-zhenwu** [milvus-io/milvus#3400]
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```
[2020-08-22 12:17:58,821][INFO][SERVER][Flush][grpcpp_sync_ser] Request [52] Flush begin.
[2020-08-22 12:17:59,185][INFO][SERVER][Flush][grpcpp_sync_ser] Request [52] Flush end.
[2020-08-22 12:17:59,187][INFO][SERVER][CountCollection][grpcpp_sync_ser] Request [53] CountCollection begin.
[2020-08-22 12:17:59,188][INFO][SERVER][CountCollection][grpcpp_sync_ser] Request [53] CountCollection end.
[2020-08-22 12:17:59,190][INFO][SERVER][GetEntityByID][grpcpp_sync_ser] Request [54] GetEntityByID begin.
[2020-08-22 12:17:59,340][INFO][SERVER][GetEntityByID][grpcpp_sync_ser] Request [54] GetEntityByID end.
[2020-08-22 12:17:59,342][INFO][SERVER][CountCollection][grpcpp_sync_ser] Request [55] CountCollection begin.
[2020-08-22 12:17:59,342][INFO][SERVER][CountCollection][grpcpp_sync_ser] Request [55] CountCollection end.
[2020-08-22 12:17:59,348][INFO][SERVER][Search][grpcpp_sync_ser] Request [56] Search begin.
[2020-08-22 12:17:59,350][INFO][SERVER][HandleSignal][taskloader_th] Server received critical signal: 11
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] Call stack:
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0xd8ce96]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0xd8da00]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /lib64/libc.so.6(+0x36400) [0x7fa5f56ab400]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x8c21f7]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x932bce]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x5feb9d]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x5ff0b0]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x5ff241]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x46d75a]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x46eeca]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0x463c7b]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /var/lib/milvus/bin/milvus_server() [0xd9accf]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /lib64/libpthread.so.0(+0x7ea5) [0x7fa5f66daea5]
[2020-08-22 12:17:59,351][INFO][SERVER][PrintStacktrace][taskloader_th] /lib64/libc.so.6(clone+0x6d) [0x7fa5f57738dd]
[2020-08-22 12:19:05,756][INFO][SERVER][HandleSignal][milvus_server] Server received critical signal: 15
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] Call stack:
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] /var/lib/milvus/bin/milvus_server() [0xd8ce96]
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] /var/lib/milvus/bin/milvus_server() [0xd8da00]
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] /lib64/libc.so.6(+0x36400) [0x7fa5f56ab400]
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] /lib64/libpthread.so.0(pause+0x2d) [0x7fa5f66e1f5d]
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] /var/lib/milvus/bin/milvus_server() [0x42396b]
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] /lib64/libc.so.6(__libc_start_main+0xf5) [0x7fa5f5697555]
[2020-08-22 12:19:05,756][INFO][SERVER][PrintStacktrace][milvus_server] /var/lib/milvus/bin/milvus_server() [0x436897]

```

**Steps/Code to reproduce behavior**
4 collections created, and 4 threads created, run `insert/delete/search/flush/..` random operation in each collection with one thread.
```
12:17:56  2020-08-22:04:17:56,465 DEBUG    [client.py:290] Row count: 100000 in collection: <collection_1_hroe3kw4>
12:17:56  2020-08-22:04:17:56,480 INFO     [client.py:36] Milvus insert run in 0.01s
12:17:56  2020-08-22:04:17:56,731 INFO     [client.py:36] Milvus flush run in 0.25s
12:17:56  2020-08-22:04:17:56,732 DEBUG    [client.py:290] Row count: 100056 in collection: <collection_1_hroe3kw4>
12:17:56  2020-08-22:04:17:56,733 INFO     [k8s_runner.py:818] collection_1_hroe3kw4
12:17:56  2020-08-22:04:17:56,733 INFO     [k8s_runner.py:819] delete_rand
12:17:56  2020-08-22:04:17:56,734 DEBUG    [client.py:290] Row count: 100056 in collection: <collection_1_hroe3kw4>
12:17:56  2020-08-22:04:17:56,734 INFO     [client.py:194] collection_1_hroe3kw4: length to delete: 99
12:17:56  2020-08-22:04:17:56,743 DEBUG    [client.py:160] Reset length: 56
12:17:56  2020-08-22:04:17:56,745 INFO     [client.py:36] Milvus delete run in 0.0s
12:17:57  2020-08-22:04:17:57,275 INFO     [client.py:36] Milvus flush run in 0.53s
12:17:57  2020-08-22:04:17:57,276 DEBUG    [client.py:290] Row count: 99999 in collection: <collection_1_hroe3kw4>
12:17:57  2020-08-22:04:17:57,276 INFO     [client.py:198] collection_1_hroe3kw4: count after delete: 99999
12:17:57  2020-08-22:04:17:57,475 DEBUG    [client.py:290] Row count: 99999 in collection: <collection_1_hroe3kw4>
12:17:57  2020-08-22:04:17:57,476 INFO     [k8s_runner.py:818] collection_1_hroe3kw4
12:17:57  2020-08-22:04:17:57,476 INFO     [k8s_runner.py:819] compact
12:17:58  2020-08-22:04:17:58,759 INFO     [client.py:36] Milvus compact run in 1.28s
12:17:58  2020-08-22:04:17:58,760 INFO     [k8s_runner.py:818] collection_0_sbobaedz
12:17:58  2020-08-22:04:17:58,760 INFO     [k8s_runner.py:819] delete_rand
12:17:58  2020-08-22:04:17:58,762 DEBUG    [client.py:290] Row count: 100000 in collection: <collection_0_sbobaedz>
12:17:58  2020-08-22:04:17:58,762 INFO     [client.py:194] collection_0_sbobaedz: length to delete: 89
12:17:58  2020-08-22:04:17:58,806 INFO     [client.py:36] Milvus delete run in 0.0s
12:17:59  2020-08-22:04:17:59,172 INFO     [client.py:36] Milvus flush run in 0.37s
12:17:59  2020-08-22:04:17:59,174 DEBUG    [client.py:290] Row count: 99911 in collection: <collection_0_sbobaedz>
12:17:59  2020-08-22:04:17:59,174 INFO     [client.py:198] collection_0_sbobaedz: count after delete: 99911
12:17:59  2020-08-22:04:17:59,329 DEBUG    [client.py:290] Row count: 99911 in collection: <collection_0_sbobaedz>
12:17:59  2020-08-22:04:17:59,329 INFO     [k8s_runner.py:818] collection_1_hroe3kw4
12:17:59  2020-08-22:04:17:59,329 INFO     [k8s_runner.py:819] query_rand
12:17:59  2020-08-22:04:17:59,330 INFO     [client.py:276] collection_1_hroe3kw4, Search nq: 7, top_k: 32, nprobe: 12
12:19:36  RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
12:19:36  	status = StatusCode.UNAVAILABLE
12:19:36  	details = "Socket closed"
12:19:36  	debug_error_string = "{"created":"@1598069975.915026209","description":"Error received from peer ipv4:10.105.60.214:19530","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Socket closed","grpc_status":14}"
12:19:36  >
12:19:36  	{'API start': '2020-08-22 04:17:59.330589', 'RPC start': '2020-08-22 04:17:59.330610', 'RPC error': '2020-08-22 04:19:35.915650'}
12:19:36  2020-08-22:04:19:35,915 ERROR    [grpc_handler.py:58] RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
12:19:36  	status = StatusCode.UNAVAILABLE
12:19:36  	details = "Socket closed"
12:19:36  	debug_error_string = "{"created":"@1598069975.915026209","description":"Error received from peer ipv4:10.105.60.214:19530","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Socket closed","grpc_status":14}"
12:19:36  >
12:19:36  	{'API start': '2020-08-22 04:17:59.330589', 'RPC start': '2020-08-22 04:17:59.330610', 'RPC error': '2020-08-22 04:19:35.915650'}
12:19:36  2020-08-22:04:19:35,916 ERROR    [main.py:70] <_MultiThreadedRendezvous of RPC that terminated with:
12:19:36  	status = StatusCode.UNAVAILABLE
12:19:36  	details = "Socket closed"
12:19:36  	debug_error_string = "{"created":"@1598069975.915026209","description":"Error received from peer ipv4:10.105.60.214:19530","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Socket closed","grpc_status":14}"
12:19:36  >
12:19:36  2020-08-22:04:19:35,919 ERROR    [main.py:71] Traceback (most recent call last):
12:19:36    File "main.py", line 68, in queue_worker
12:19:36      runner.run(run_type, collection)
12:19:36    File "/home/jenkins/agent/workspace/milvus-benchmark/milvus_benchmark/k8s_runner.py", line 821, in run
12:19:36      task_run()
12:19:36    File "/home/jenkins/agent/workspace/milvus-benchmark/milvus_benchmark/client.py", line 284, in query_rand
12:19:36      self.query(vector_query)
12:19:36    File "/home/jenkins/agent/workspace/milvus-benchmark/milvus_benchmark/client.py", line 259, in query
12:19:36      result = self._milvus.search(self._collection_name, query)
12:19:36    File "/usr/local/lib/python3.6/site-packages/milvus/client/stub.py", line 35, in inner
12:19:36      return func(self, *args, **kwargs)
12:19:36    File "/usr/local/lib/python3.6/site-packages/milvus/client/stub.py", line 614, in search
12:19:36      return handler.search(collection_name, query_entities, partition_tags, fields, timeout=timeout, **kwargs)
12:19:36    File "/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py", line 59, in handler
12:19:36      raise e
12:19:36    File "/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py", line 42, in handler
12:19:36      return func(self, *args, **kwargs)
12:19:36    File "/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py", line 552, in search
12:19:36      response = ft.result()
12:19:36    File "/usr/local/lib/python3.6/site-packages/grpc/_channel.py", line 625, in result
12:19:36      raise self
12:19:36  grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
12:19:36  	status = StatusCode.UNAVAILABLE
12:19:36  	details = "Socket closed"
12:19:36  	debug_error_string = "{"created":"@1598069975.915026209","description":"Error received from peer ipv4:10.105.60.214:19530","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Socket closed","grpc_status":14}"
12:19:36  >
12:19:36  
12:20:44  2020-08-22:04:20:35,966 DEBUG    [k8s_runner.py:72] Start clean up: milvus-benchmark-test-ppaia0z8
12:20:44  2020-08-22:04:20:35,966 DEBUG    [utils.py:386] helm uninstall -n milvus milvus-benchmark-test-ppaia0z8
12:20:44  release "milvus-benchmark-test-ppaia0z8" uninstalled
```

**Expected behavior**

**Method of installation**
- [x] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version (master or released version)

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

**Comment by BossZou** [comment]
Fixed in #3436

**Comment by shengjun1985** [comment]
duplicate with #3330

**Comment by BossZou** [comment]
Fixed by #3436
